internal/db/collection_delete.go

```diff
@@ -148,15 +151,15 @@ func (c *collection) applyDelete(
 		return err
 	}
 
-	dagLinks := make([]core.DAGLink, len(cids))
+	dagLinks := make([]coreblock.DAGLink, len(cids))
```

Comment:
thought: It feels like this is leaking abstractions/dependencies - it might be good to move this into the `merkle` pkg or similar at some point

Reply:
Can you explain why you're saying this? I don't think the DAGLink should move to the `merkle` package.

Reply:
It feels out of place to have the `db` package mess around directly with DAGLinks, and it is only done here (and on save).  The way they are interacted with looks easy to move out of the package too

Reply:
Ah I see what you mean. I tried that before but I don't think it's possible to avoid it given that on save we need to keep track of the link names. It would be weird to build a different array only to convert it in the `merkleCRDT` package.